### PR TITLE
Fixed #25547 -- Corrected instance assignment in FileDescriptor.__get__

### PR DESCRIPTION
**Pull request by Uran198**

### CODE REVIEW

File: django/db/models/fields/files.py [hunk 1]
```diff
@@ -201,6 +201,9 @@ def __get__(self, instance, cls=None):
             file.instance = instance
             file.field = self.field
             file.storage = self.field.storage
+        # Also, make sure that instance is correct.
+        elif isinstance(file, FieldFile) and id(instance) != id(file.instance):
```
Comment (charettes) [inline django/db/models/fields/files.py]: Wouldn't `instance is file.instance` work here?

File: tests/model_fields/tests.py [hunk 2]
```diff
@@ -792,6 +792,11 @@ def test_delete_when_file_unset(self):
         except OSError:
             self.fail("Deleting an unset FileField should not raise OSError.")
 
+    def test_refresh_from_db(self):
+        d = Document.objects.create(myfile='something.txt')
+        d.refresh_from_db()
+        self.assertEqual(id(d.myfile.instance), id(d))
```
Comment (charettes) [inline tests/model_fields/tests.py]: I think there's an `assertIs` method.

File: django/db/models/fields/files.py (hunk 1 above)
Comment (Uran198) [inline django/db/models/fields/files.py]: Yes, it would, thanks.

File: tests/model_fields/tests.py (hunk 2 above)
Comment (Uran198) [inline tests/model_fields/tests.py]: Yes, there is, thanks.

**Comment by timgraham** [comment]
merged in 6f229048ddd8c7347ff60dddfb9121e6021c7b2e, thanks!

**Pull request closed without merging**